test/disable-masterkey-test.js

```diff
@@ -0,0 +1,121 @@
+let assert = require('assert-diff');
+let async = require('async');
+let Remote = require('ripple-lib').Remote;
+let testutils = require('./testutils');
+
+let {alice,bob,carol,dan,root,mtgox,bitstamp,amazon} = require('./testconfig').accounts;
+
+suite('Disabling MasterKey', function() {
+  let R = {};
+  let $ = {helpers: R};
+
+  setup(function(done) {
+    testutils.build_setup({local_signing: true}).call($, done);
+  });
+
+  teardown(function(done) {
+    testutils.build_teardown().call($, done);
+  });
+
+  test('lsfPasswordSpent', function*() {
+    let password_spent_flag = Remote.flags.account_root.PasswordSpent;
+    let regular_key = 'rGLnRYhy5fQK5pxZuMxtsJKrbu5onBpRst';
+
+    yield R.create_account(root, alice);
+
+    assert(!((yield R.get_account_info(alice)).Flags & password_spent_flag),
+           'PasswordSpent flag set unexpectedly');
+
+    yield R.promise(function(callback) {
+      let tx = $.remote.createTransaction('SetRegularKey', {
+        account: alice,
+        regular_key: regular_key
+      });
+
+      // As a special feature, each account is allowed to perform
+      // SetRegularKey transaction without a transaction fee as long as
+      // the lsfPasswordSpent flag for the account is not set
+      //
+      // https://github.com/ripple/rippled/blob/4239880acb5e559446d2067f00dabb31cf102a23/src/ripple/app/transactors/SetRegularKey.cpp#L64-L67
+      tx.setFixedFee(0);
+
+      tx.once('submitted', function(res) {
+        assert.strictEqual(res.engine_result, 'tesSUCCESS');
+      });
+
+      $.submit_transaction(tx, callback);
+    });
+
+    assert(((yield R.get_account_info(alice)).Flags & password_spent_flag),
+           'PasswordSpent flag not set');
+
+    yield R.promise(function(callback) {
+      // The second SetRegularKey transaction with Fee=0 should fail.
+      // The initial engine_result is telINSUF_FEE_P. ripple-lib waits for the
+      // transaction's LastLedgerSequence to expire, so the final result is a
+      // locally-determined failure: tejMaxLedger
+      let tx = $.remote.createTransaction('SetRegularKey', {
+        account: alice,
+        regular_key: regular_key
+      });
+
+      tx.setFixedFee(0);
+      tx.once('submitted', function(res) {
+        assert.strictEqual(res.engine_result, 'telINSUF_FEE_P');
+      });
+
+      $.submit_transaction(tx, function(err) {
+        assert.strictEqual(err.result, 'tejMaxLedger');
+        callback();
+      });
+    });
+
+    yield R.payment(root, alice, '1');
+
+    assert(!((yield R.get_account_info(alice)).Flags & password_spent_flag),
+           'PasswordSpent flag set unexpectedly');
+  });
+
+  test('Disable master key', function*() {
```

Comment:
A negative test case, where the `AccountSet` with `asfDisableMaster` is attempted without a regular key, would probably be useful here.